CONTRIBUTING.md

```diff
@@ -0,0 +1,206 @@
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+- **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+- **Describe the current behavior and explain which behavior you expected to see instead** and why.
+- **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of the project which the suggestion is related to.
+- **Explain why this enhancement would be useful** to our users.
+
+### **First Code Contribution 🥇**
+
+Unsure on where to begin your contribution? You can start by looking at issues having `Good First Issues`, `Documentation` or `Beginner` tags. You can fine theme [here](https://github.com/kachun333/OurPromise/issues)
+
+### **Pull Request 🥳**
+
+Once you have [forked](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the repo, you need to create your code contributions within a new branch of your forked repo.  
+For general background on creating and managing branches within GitHub, see: [Git Branching and Merging](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging).
+
+Following are the steps to guide you:
+
+**Step 0:** Clone the forked repo and cd into it
+
+```
+git clone https://github.com/<Your username>/OurPromise.git
+cd OurPromise/
+```
+
+**Step 1:** To begin, create a new branch, give a good [branch name](https://codingsight.com/git-branching-naming-convention-best-practices/). You usually create a branch like so:
+
+```
+git checkout master
+git checkout -b [name_of_your_new_branch]
+```
+
+**Step 2:** Now make your changes and then stage your files for commit.
+
+```
+git add [File name]    --> To stage one particular files
+or
+git add .              --> To stage all the modified files
+```
+
+**Step 3:** Commit all the changes. The [commit message should be meaningful](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/), short and to the point.
+
+```
+git commit -m "Write a meaningfull, small commit message"
+```
+
+**Step 4:** Now push the changes for review
+
+```
+git push origin <name_of_your_new_branch>
+```
+
+**Step 5:** After you push your code, github would detect the new branch and an option would be there to create a pull request.
+
+**Step 6:** The last step is to give a detailed description and short title in your pull request.
+
+- **Dont forget** to [link your PR with the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) using specific keywords.
+- Provide a **screen shot** if necessary.
+- Give a **detailed description** on what you changed.
```

Comment:
[Note]
this is good for now, but we will need a Pull Request Template to standardize things
Issue opened at #52